Sources/Network/BoxNetworkAgent.swift

```diff
@@ -143,7 +143,24 @@ public class BoxNetworkAgent: NSObject, NetworkAgentProtocol {
                 completion: completion
             )
         }
-        task.resume()
+
+        // Key value observer: Observer attaches to Progress object on task. Every time the Progress object updates,
+        // progressHandler is called with the Progress object passed in. progressHandler calls the request's progress closure and passes in the updated
+        // progress object.
+        var observation: NSKeyValueObservation?
+        let progressHandler: (Progress, NSKeyValueObservedChange<Double>) -> Void = {
+            progress, _ in
+            request.progress(progress)
+            if progress.fractionCompleted >= 1 {
+                observation?.invalidate()
+                observation = nil
```

Comment:
I wouldn't block on this, but I think it'll be worth testing the case that a task is cancelled before completion, to ensure that the observation is cleaned up.

Reply:
Made a change where I invalidate the observation in callback of the URLSessionTask. So in the case where the task is cancelled, the callback will be called with an error and the observation will be invalidated.